### ISSUE TITLE
Define a property to identify the SDK version

### ISSUE THREAD

**Issue by livarcocc** [dotnet/cli#7051]
_From @AArnott on December 19, 2016 5:2_

The cross-targeting / SDK project type offer new functionality that a build extension may want to detect in order to reliably leverage (as opposed to setting properties and blindly hoping the new functionality is present).

Please define a new property that a build extension can check for to detect the use of the new SDK attribute in the project file.

Per a brief side discussion on #528 with @nguerrera.

_Copied from original issue: dotnet/sdk#534_

**Comment by livarcocc** [comment]
_From @onovotny on December 21, 2016 16:49_

More than just a property, we need to know how to call it when referring to it in documentation. It needs a "marketing name" too.

**Comment by livarcocc** [comment]
_From @dsplaisted on December 30, 2016 1:5_

I'd suggest defining a property that indicates the version number of the .NET SDK that is being used.  That way if a NuGet package depends on functionality in a new version of an SDK, the package can have MSBuild logic that will generate an error saying you need to use a newer SDK if the current version isn't high enough.

**Comment by livarcocc** [comment]
_From @nguerrera on December 30, 2016 22:30_

@dsplaisted Makes sense. Which version should we use? I think it should be System.Version parse-able to be friendly to msbuild's support for semantically correct comparisons of those.

**Comment by livarcocc** [comment]
_From @dsplaisted on January 6, 2017 22:59_

@nguerrera yes, I think it should be a System.Version version, and it would be nice to have another property indicating the semantic version suffix (ie "alpha-1234" or whatever)

**Comment by livarcocc** [comment]
_From @ericstj on January 7, 2017 0:44_

+1 to this.  I'm actually blocked on this.  I'm creating a nuget package that needs to understand if its in nuget2 vs nuget3 vs nuget4 mode.  I thought I had a good condition for this: https://github.com/dotnet/standard/commit/09e9f9ecda4aa9eeedc2aadbd1b0bdaff8d6e010#diff-de762529dcf068a5867280f815198927R9 but it looks like the properties I was using are actually set by targets imported *after* those imported by NuGet packages so my condition doesn't work 😕.  Can you please set a property in `.props` that indicates the project is in PackageReference mode?

**Comment by livarcocc** [comment]
_From @ericstj on May 11, 2017 16:55_

+1 again.  I'd like to poison our new framework packages and point folks to update their SDK.  I'd really like something explicit to trigger that on, rather than a side-effect of implementation.

**Comment by livarcocc** [comment]
_From @AArnott on May 11, 2017 17:14_

Maybe if folks +1 the top-level issue itself it will gain more visibility. :)

**Comment by livarcocc** [comment]
_From @nguerrera on May 11, 2017 17:26_

@dsplaisted can you poach this as part of your bug push?

**Comment by livarcocc** [comment]
_From @ericstj on May 11, 2017 18:40_

Ideally this is a property to identify both the SDK project type and the SDK version.

**Comment by livarcocc** [comment]
_From @nguerrera on May 11, 2017 18:50_

What do you mean by sdk project type?

**Comment by livarcocc** [comment]
_From @ericstj on May 11, 2017 18:51_

What @AArnott described initially.

**Comment by livarcocc** [comment]
_From @nguerrera on May 11, 2017 20:56_

So "is this an sdk project or not" not "what kind of sdk project is this". Got it.

**Comment by livarcocc** [comment]
_From @nguerrera on May 23, 2017 21:22_

Implementing the boolean part in #1242. Plumbing the version through is more problematic so keeping that separate.

**Comment by livarcocc** [comment]
Closing this issue in the SDK side and moving it to the CLI side to provide an SDK version property. We should do something similar to Microsoft.NETCoreSdk.BundledVersions.props.

**Comment by AArnott** [comment]
So moving this issue to dotnet/cli shouldn't mean that only `dotnet build` invocations will see the new property, will it? I need it for MSBuild.exe too.

**Comment by nguerrera** [comment]
No, it's an implementation detail of how we generate version props. It'll be available to msbuild too.

**Comment by davidfowl** [comment]
@nguerrera looks like this is blocking https://github.com/dotnet/project-system/issues/2248 any idea how much work this is?

**Comment by nguerrera** [comment]
Pretty small now that we had the insight to do it as bundled runtime versions are implemented.

**Comment by nguerrera** [comment]
Alternative implementation might be for msbuild to surface an api to get versions of resolved SDKs. We return a version in the resolver. That's more return than this but would work for older sdks.

Hacky alternative would be to check for imported Sdk.props path in project system (they know to reload when targets change so I assume they can get this) and show name of ..\..\.. if it parses as semantic version. That's putting knowledge of sdk implementation detail in project system, though, so I'd much rather use this feature or an msbuild one.
